docs/developer-docs/security/security-best-practices/data-integrity-and-authenticity.mdx

```diff
@@ -12,16 +12,25 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
 
 <MarkdownChipRow labels={["Advanced", "Security", "Concept"]} />
 
-## Using certified variables for secure queries 
+## Certified variables
 
-ICP offers three modes of operations for canisters, `update`, `query` and `composite_query`. For the sake of simplicity, we will club `composite_query` under queries for the rest of this section. For more information, view the  [detailed overview between `update` and `query` calls](https://internetcomputer.org/docs/current/developer-docs/smart-contracts/call/overview/#introduction).
+### Security concern
+
+ICP offers three modes of operations for canisters, `update`, `query` and `composite_query`. For the sake of simplicity, we will club `composite_query` under queries for the rest of this section. For more information, view the  [detailed overview between `update` and `query` calls](https://internetcomputer.org/docs/current/developer-docs/smart-contracts/call/overview/#introduction). As explained in the overview, update calls are slow and expensive but provide integrity guarantee as they are responded with a threshold signature signed by the subnet. 
+
+On the other hand, query calls are fast since they are responded by a single replica but **there is no integrity guarantee, since the response can be manipulated if the responding replica or the boundary node is malicious.** For example, if the NNS dapp fetches proposal information from the governance canister via query calls and the responding node is malicious, it can mask an ill intentioned proposal that causes irrevocable damage as innocuous by modifying the proposal payload in the response and mislead voters into voting yes. Another consequence of query calls is that users can't rely on [canister_inspect_message](https://internetcomputer.org/docs/current/developer-docs/security/security-best-practices/iam#do-not-rely-on-ingress-message-inspection) as a guard. **This makes query calls, in it's raw form, unfit to serve data for security critical applications.**
```

Comment:
```suggestion
On the other hand, query calls are fast since a single replica formulates the response but **there is no integrity guarantee, since the response can be manipulated if the responding replica or the boundary node is malicious.** For example, if the NNS dapp fetches proposal information from the governance canister via query calls and the responding node is malicious, it can mask an ill intentioned proposal that causes irrevocable damage as innocuous by modifying the proposal payload in the response and mislead voters into voting yes. Another consequence of query calls is that users can't rely on [canister_inspect_message](https://internetcomputer.org/docs/current/developer-docs/security/security-best-practices/iam#do-not-rely-on-ingress-message-inspection) as a guard. **This makes query calls, in it's raw form, unfit to serve data for security critical applications.**
```

Inspect message should never be relied upon for anything but DoS protection. I'm not sure if your phrasing suggests otherwise. Do you mean inspect message only runs on update calls or that query calls have no other method for access control?